### PR TITLE
Apply misc. improvements to "Get started" documentation

### PR DESCRIPTION
**Pull request by Viicos**
## Change Summary

Mostly wording, outdated things and crossrefs. I'm planning on tackling some sections where I believe more rework could be done

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -43,14 +41,14 @@ Pydantic has few dependencies, doesn't require compiling and tests don't need ac
...
-    **tl;dr**: use `make format` to fix formatting, `make` to run tests and linting & `make docs`
+    **tl;dr**: use `make format` to fix formatting, `make` to run tests and linting and `make docs`
     to build the docs.
 
 ### Prerequisites
 
 You'll need the following prerequisites:
 
-- Any Python version between **Python 3.8 and 3.11**
+- Any Python version between **Python 3.9 and 3.12**
```
Comment (Viicos) [inline docs/contributing.md:51]: Figured we should recommend at least 3.9 as we might have some dev dependencies requiring 3.9 (was able to remove the one we were talking to yesterday, but might happen in the future).

File: docs/contributing.md [hunk 2]
```diff
@@ -14,8 +14,6 @@ If you're using Pydantic prior to **v2.0** please use:
 ```bash
 python -c "import pydantic.utils; print(pydantic.utils.version_info())"
 ```
-If you're using Pydantic prior to **v1.3** (when `version_info()` was added), please manually include OS, Python
-version and pydantic version.
```
Comment (Viicos) [inline docs/contributing.md:18]: Removed as 1.3 seems pretty old now.

File: docs/contributing.md [hunk 3]
```diff
@@ -124,9 +122,6 @@ Please follow the pull request template and fill in as much information as possi
 
 When your pull request is ready for review, add a comment with the message "please review" and we'll take a look as soon as we can.
 
-## Code style and requirements
-
-TODO
```
Comment (Viicos) [inline docs/contributing.md:129]: Removed as I don't think there's much to be put in this section.

**Review by sydney-runkle — APPROVED** [review]
Nice work! Feel free to merge once linting issues are fixed.

**Pull request merged**